src/_shared/_constants/trackGroups.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ENetwork } from '@shared/types';
+import { NETWORKS_DETAILS } from './networks';
+
+export type TrackGroup = 'Main' | 'Treasury' | 'Whitelist' | 'Governance' | 'Origin';
+
+/**
+ * Dynamically generates track groups based on network track details and track purposes
+ * @param network - The network to get track groups for
+ * @returns Object with track groups and their associated track IDs
+ */
+export function getTrackGroups(network: ENetwork): Record<TrackGroup, number[]> {
+	const trackDetails = NETWORKS_DETAILS[network]?.trackDetails;
+
+	if (!trackDetails) {
+		return {
+			Main: [],
+			Treasury: [],
+			Whitelist: [],
+			Governance: [],
+			Origin: []
+		};
+	}
+
+	const groups: Record<TrackGroup, number[]> = {
+		Main: [],
+		Treasury: [],
+		Whitelist: [],
+		Governance: [],
+		Origin: []
+	};
+
+	// Group tracks by their actual purpose and functionality
+	Object.values(trackDetails).forEach((track) => {
+		if (track && typeof track.trackId === 'number') {
+			const { trackId } = track;
+			const trackName = track.name;
```

Comment:
use the key, because you can use Enums EPostOrigin for it, don't hard code and validate for trackNames